pkgs/applications/audio/renoise/default.nix

```diff
@@ -12,21 +12,31 @@ in
 
 stdenv.mkDerivation rec {
   pname = "renoise";
-  version = "3.3.2";
-
-  src =
-    if stdenv.hostPlatform.system == "x86_64-linux" then
-        if releasePath == null then
-        fetchurl {
-          urls = [
-              "https://files.renoise.com/demo/Renoise_${urlVersion version}_Demo_Linux.tar.gz"
-              "https://web.archive.org/web/https://files.renoise.com/demo/Renoise_${urlVersion version}_Demo_Linux.tar.gz"
-          ];
-          sha256 = "0d9pnrvs93d4bwbfqxwyr3lg3k6gnzmp81m95gglzwdzczxkw38k";
-        }
-        else
-          releasePath
-    else throw "Platform is not supported. Use installation native to your platform https://www.renoise.com/";
+  version = "3.4.3";
+
+  src = if stdenv.hostPlatform.system != "x86_64-linux" then
+    throw
+    "Platform is not supported. Use installation native to your platform https://www.renoise.com/"
+  else if releasePath != null then
+    releasePath
+  else
+    let
+      mkUrl = { urlStem, archSuffix }:
+        lib.strings.concatStrings [
+          urlStem
+          "https://files.renoise.com/demo/Renoise_"
+          (urlVersion version)
+          "_Demo_Linux"
+          archSuffix
+          ".tar.gz"
+        ];
+    in fetchurl {
+      urls = map mkUrl (lib.attrsets.cartesianProductOfSets {
```

Comment:
So the idea here is to allow a user to override `version` and `src.hash` and still get a coherent derivation. An archive.org link is also important because it’s going to keep the derivation fresh even if renoise.com decides to move on to a new version. If we don’t care about this form of backwards compat, we can rip all of it out in favor of the live URL (the first one in the list).

Reply:
…Actually what should really happen here is that the derivation should support other systems besides x86_64, which upstream now supports (the download page now also lists aarch64-linux, as well as `armhf`, which I don’t believe corresponds to any known system double). So I’m going to rewrite the logic to just do that and grab the appropriate download link for each of the two distros.